src/EditorFeatures/Core/Structure/AbstractStructureTaggerProvider.cs

```diff
@@ -67,37 +67,146 @@ protected override bool ComputeInitialTagsSynchronously(ITextBuffer subjectBuffe
+
+            if (!collapseRegions && !collapseUsings)
+            {
+                return false;
+            }
 
-            // If we contain any #region sections, we want to collapse those automatically on open the first
-            // time a doc is ever opened.  So we need to compute the initial tags synchronously in order to
-            // do that.
-            if (ContainsRegionTag(subjectBuffer.CurrentSnapshot))
+            if (ContainsRegionOrUsing(subjectBuffer.CurrentSnapshot, collapseRegions, collapseUsings, openDocument.Project.Language))
+            {
                 return true;
+            }
 
             return false;
+        }
 
-            static bool ContainsRegionTag(ITextSnapshot textSnapshot)
+        // Internal for testing. Can't use TestAccessor because the base class already defines one
+        internal static bool ContainsRegionOrUsing(ITextSnapshot textSnapshot, bool collapseRegions, bool collapseUsings, string language)
+        {
+            foreach (var line in textSnapshot.Lines)
             {
-                foreach (var line in textSnapshot.Lines)
+                if (collapseRegions && StartsWithRegionTag(line))
+                {
+                    return true;
+                }
+                else if (collapseUsings && IsUsingDeclarationOrExtern(line, language))
                 {
-                    if (StartsWithRegionTag(line))
-                        return true;
+                    return true;
                 }
+            }
+
+            return false;
+
+            static bool StartsWithRegionTag(ITextSnapshotLine line)
+            {
+                if (line.Length < 7)
+                    return false;
+
+                var index = line.Start.Position;
+                Skip(line.Snapshot, ref index, line.End.Position, whitespace: true);
+                // Ignore case here because this deals with VB and C#
+                return line.StartsWith(index, "#region", ignoreCase: true);
+            }
 
+            static bool IsUsingDeclarationOrExtern(ITextSnapshotLine line, string language)
+            {
+                if (line.Length < 8)
+                    return false;
+
+                var index = line.Start.Position;
+                var text = line.Snapshot;
+                var lineEnd = line.End.Position;
+
+                // Skip whitespace at the start of the line
+                Skip(text, ref index, lineEnd, whitespace: true);
+
+                // For VB we only need to find "Imports" at the start of a line
+                if (language == LanguageNames.VisualBasic)
+                {
+                    return line.StartsWith(index, "Imports", ignoreCase: true);
+                }
+
+                // For the purposes of collapsing, extern aliases are grouped with usings, so
+                // we need to check for them too
+                if (line.StartsWith(index, "extern alias", ignoreCase: false))
```

Comment:
😬 this totally ignores changes in spacing between the keywords

Reply:
GitHub needs a facepalm reaction